portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -0,0 +1,88 @@
+package com.google.sps.data;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+public class LongestTaskFirstScheduler implements TaskScheduler {
+
+  // Comparator for sorting tasks by duration in ascending order
+  public static final Comparator<Task> sortByTaskDurationAscending =
+      Comparator.comparing(Task::getDuration);
+
+  /**
+   * Schedules the tasks so that the longest tasks are scheduled to the first possible free time
+   * range of the day. This approach tries to prioritize long tasks so that they are scheduled.
+   */
+  public Collection<ScheduledTask> schedule(
+      Collection<CalendarEvent> events,
+      Collection<Task> tasks,
+      Instant workHoursStartTime,
+      Instant workHoursEndTime) {
+
+    List<CalendarEvent> eventsList = new ArrayList<CalendarEvent>(events);
+    List<Task> tasksList = new ArrayList<Task>(tasks);
+
+    // Sorts the tasks in descending order based on duration.
+    // Longest task comes first in the collection.
+    // Then sorts the list again by alphabetical order ascending.
+    Collections.sort(
+        tasksList,
+        Collections.reverseOrder(sortByTaskDurationAscending).thenComparing(Task::getName));
+
+    CalendarEventsGroup calendarEventsGroup =
+        new CalendarEventsGroup(eventsList, workHoursStartTime, workHoursEndTime);
+    List<TimeRange> availableTimes = calendarEventsGroup.getFreeTimeRanges();
+
+    // Create a TimeRangeGroup class for the free time ranges.
+    TimeRangeGroup availableTimesGroup = new ArrayListTimeRangeGroup(availableTimes);
+
+    List<ScheduledTask> scheduledTasks = new ArrayList<ScheduledTask>();
+
+    for (Task task : tasksList) {
+      Duration taskDuration = task.getDuration();
+
+      // Find the first free time range that is longer than the current task's duration.
+      for (TimeRange currentFreeTimeRange : availableTimes) {
+        if (currentFreeTimeRange.duration().compareTo(taskDuration) >= 0) {
```

Comment:
I think it'd be simpler to continue in the opposite case.

ex:
if (currentFreeTimeRange.duration().compareTo(taskDuration) < 0) {
  continue;
}

Reply:
Updated.